openmc/mgxs/mdgxs.py

```diff
@@ -0,0 +1,2619 @@
+            return (0, 1, 3, 4)
+        else:
+            return (1, 2)
+
+    @property
+    def delayed_groups(self):
+        return self._delayed_groups
+
+    @property
+    def num_delayed_groups(self):
+        if self.delayed_groups is None:
+            return 1
+        else:
+            return len(self.delayed_groups)
+
+    @delayed_groups.setter
+    def delayed_groups(self, delayed_groups):
+
+        if delayed_groups is not None:
+
+            cv.check_type('delayed groups', delayed_groups, list, int)
+            cv.check_greater_than('num delayed groups', len(delayed_groups), 0)
+
+            # Check that the groups are within [1, MAX_DELAYED_GROUPS]
+            for group in delayed_groups:
+                cv.check_greater_than('delayed group', group, 0)
+                cv.check_less_than('delayed group', group, MAX_DELAYED_GROUPS,
+                                   equality=True)
+
+        self._delayed_groups = delayed_groups
+
+    @property
+    def filters(self):
+
+        # Create the non-domain specific Filters for the Tallies
+        group_edges = self.energy_groups.group_edges
+        energy_filter = openmc.EnergyFilter(group_edges)
+
+        if self.delayed_groups != None:
+            delayed_filter = openmc.DelayedGroupFilter(self.delayed_groups)
+            filters = [[energy_filter], [delayed_filter, energy_filter]]
+        else:
+            filters = [[energy_filter], [energy_filter]]
+        return self._add_angle_filters(filters)
+
+    @staticmethod
+    def get_mgxs(mdgxs_type, domain=None, domain_type=None, energy_groups=None,
+                 delayed_groups=None, by_nuclide=False, name='',
+                 num_polar=1, num_azimuthal=1):
+        """Return a MDGXS subclass object for some energy group structure within
+        some spatial domain for some reaction type.
+
+        This is a factory method which can be used to quickly create MDGXS
+        subclass objects for various reaction types.
+
+        Parameters
+        ----------
+        mdgxs_type : {'delayed-nu-fission', 'chi-delayed', 'beta', 'decay-rate', 'delayed-nu-fission matrix'}
+            The type of multi-delayed-group cross section object to return
+        domain : openmc.Material or openmc.Cell or openmc.Universe or
+            openmc.RegularMesh
+            The domain for spatial homogenization
+        domain_type : {'material', 'cell', 'distribcell', 'universe', 'mesh'}
+            The domain type for spatial homogenization
+        energy_groups : openmc.mgxs.EnergyGroups
+            The energy group structure for energy condensation
+        by_nuclide : bool
+            If true, computes cross sections for each nuclide in domain.
+            Defaults to False
+        name : str, optional
+            Name of the multi-group cross section. Used as a label to identify
+            tallies in OpenMC 'tallies.xml' file. Defaults to the empty string.
+        delayed_groups : list of int
```

Comment:
Question: Is this also optional (it is defaulted to `None`).  If so, it should be noted here similar to the other "optional" parameters.  If it is not optional, it is more standard to have non-optional parameters precede optional parameters.